rwengine/src/ai/CharacterController.cpp

```diff
@@ -40,8 +40,10 @@ void CharacterController::skipActivity() {
     // vehicle
     // or jumping.
     if (getCurrentActivity() != nullptr &&
-        getCurrentActivity()->canSkip(character, this))
+        getCurrentActivity()->canSkip(character, this)) {
```

Comment:
I know this is not your code addition, but I think while you are at it, you should introduce a temporary variable to avoid checking a different currentActivity than what's actually used in the canSkip().
We never know what kind of code we'll add, but I think with network play or something in the future (callbacks etc.), we probably want this safer and it's an easy addition :)